job/server/src/main/java/alluxio/job/JobWorkerContext.java

```diff
@@ -20,23 +22,44 @@
  */
 @ThreadSafe
 public final class JobWorkerContext {
+  private final FileSystem mFileSystem;
+  private final FileSystemContext mFsContext;
   private final long mJobId;
   private final int mTaskId;
   private final UfsManager mUfsManager;
 
   /**
    * Creates a new instance of {@link JobWorkerContext}.
    *
+   * @param filesystem the filesystem client that is used when running jobs
+   * @param fsContext the filesystem client's underlying context
    * @param jobId the job id
    * @param taskId the task id
    * @param ufsManager the UFS manager
    */
-  public JobWorkerContext(long jobId, int taskId, UfsManager ufsManager) {
+  public JobWorkerContext(FileSystem filesystem, FileSystemContext fsContext, long jobId,
+      int taskId, UfsManager ufsManager) {
```

Comment:
Why does the `JobWorkerContext` have a `jobId` and `taskId`? Is this more of a `JobTaskContext`?

Reply:
see explanation from `JobMasterContext`, but essentially something like that. maybe `RunTaskContext` might be more appropriate